File: Lab6/Question7/src/Dot.java
public class Dot extends Shape{
    public Dot(double x, double y) {
        super(x, y);
    }

    @Override
    public double area() {
        return 0;
    }

    public static void testDot() {
        Dot dot = new Dot(1, 2);
        //Test attributes of dot
        System.out.println(dot.getX() == 1);
        System.out.println(dot.getY() == 2);
        System.out.println(dot.area() == 0);
    }
    
    @Override
    public String toString() {
        return this.getClass().getName() + " has area " + area();
    }
}

File: Lab6/Dot.kt
class Dot : Shape {
    constructor(x: Double, y: Double) : super(x, y)

    override fun area(): Double {
        return 0.0
    }

    override fun toString(): String {
        return "${this.javaClass.name} has area ${area()}"
    }

    companion object {
        fun testDot() {
            val dot = Dot(1.0, 2.0)
            println(dot.getX() == 1.0)
            println(dot.getY() == 2.0)
            println(dot.area() == 0.0)
        }
    }
}